### PR TITLE
Added types for plotly.js-basic-dist

### PR DESCRIPTION
**Pull request by DreierF**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

Links:
- https://www.npmjs.com/package/plotly.js-basic-dist
- https://github.com/plotly/plotly.js/blob/master/dist/README.md#plotlyjs-basic

### CODE REVIEW

**Comment by DreierF** [comment]
> 🔔 @DreierF — you're the only owner, but it would still be good if you find someone to review this PR in the next few days, otherwise a maintainer will look at it. (And if you do find someone, maybe even recruit them to be a second owner to make future changes easier...)

@chrisgervang @martinduparc @frederikaalund @taoqf @Dadstart @szechyjs @soorajpudiyadath @jonfreedman @meganrm @milesjos @SkipperCool @mtadams007 @marnett-git @peterblazejewicz @brammitch @blizzardjessica @olegshilov @PabloGracia 

As all of you contributed to the plotly.js package, could someone of you review this change?

File: types/plotly.js-basic-dist/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for plotly.js-basic-dist 1.54
+// Project: https://github.com/plotly/plotly.js#readme
+// Definitions by: Florian Dreier <https://github.com/me>
```
Comment (peterblazejewicz) [inline types/plotly.js-basic-dist/index.d.ts:3]:
```suggestion
// Definitions by: Florian Dreier <https://github.com/DreierF>
```

**Comment by peterblazejewicz** [comment]
@DreierF does it mean only some types of traces should be re-exported from `plotly.js`? At the moment, IMO this package
is just a pass-through for all `plotly.js` definitions.

**Comment by DreierF** [comment]
Thanks for the review!
Yes, ideally only the relevant subset should be re-exported, but it would be an enormous effort to cherry-pick only the relevant types. WDYT?

I took inspiration of the [jQuery slim package](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jquery/dist/jquery.slim.d.ts), which also just exposes a subset of jQuery in reality, but exports all the types.

**Review by peterblazejewicz — APPROVED** [review]
@DreierF agree, LGTM!
thx!

**Comment by DreierF** [comment]
Ready to merge

File: types/plotly.js-basic-dist/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for plotly.js-basic-dist 1.54
```
Comment (DreierF) [inline types/plotly.js-basic-dist/index.d.ts:1]: Should be `plotly.js-basic-dist-min`

File: types/plotly.js-basic-dist/plotly.js-basic-dist-tests.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import * as Plotly from 'plotly.js-basic-dist';
```
Comment (DreierF) [inline types/plotly.js-basic-dist/plotly.js-basic-dist-tests.ts:1]: Similarly here to test the typings for `plotly.js-basic-dist-min` and not those of `plotly.js-basic-dist`

File: types/plotly.js-basic-dist/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                "d3/v3"
+            ]
+        },
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts",
+        "plotly.js-basic-dist-tests.ts"
```
Comment (DreierF) [inline types/plotly.js-basic-dist/tsconfig.json:27]: File should probably be renamed accordingly as well to be consistent

**Pull request merged**